Add Diffblue Deeptest tests

petersear opened the pull request:
Pre-PR test review request.
These tests are also available to view at:
http://cx.dashboard.diffblue.com/Mybatis-PageHelper/1/tests

**PR text:**

The test cases contained in this PR were automatically generated by [Diffblue](http://www.diffblue.com/)'s [Deeptest](http://www.diffblue.com/papers/overview.pdf) software. 

Diffblue is an Oxford-based start-up that aims to automate traditional coding tasks to increase developer productivity. One of our missions is to provide useful tests to open source projects. In return we would appreciate your feedback about our tests to help us further improve our tool. 
    
These tests are part of a batch of 102 in total, providing an increase of 9% in line coverage.
We would be happy to provide the remaining tests via a subsequent PR.

Results for various open source projects can be viewed at:
    [http://diff.blue/demo](http://diff.blue/demo)

We welcome any feedback you have on these tests.

paulkennethkent commented on pom.xml:
```diff
@@ -103,6 +103,33 @@
+            <groupId>org.powermock</groupId>
+            <artifactId>powermock-api-mockito-common</artifactId>
+            <version>1.6.5</version>
+        </dependency>
```
powermock-api-mockito-common is not required, there is a PR to change this.

petersear replied:
Done, thanks.

paulkennethkent commented on src/test/java/com/github/pagehelper/dialect/AbstractRowBoundsDialectTest.java:
```diff
@@ -0,0 +1,80 @@
+   */
+
+  @Test
+  public void beforeCountInputNotNullZeroNotNullOutputFalse() throws InvocationTargetException {
```
The already have a test that covers this line according to the online demo. Might be worth leaving out.

petersear replied:
Removed.

paulkennethkent commented on src/test/java/com/github/pagehelper/dialect/AbstractRowBoundsDialectTest.java:
```diff
@@ -0,0 +1,80 @@
+   */
+
+  @Test
+  public void beforeCountInputNotNullZeroNotNullOutputTrue() throws InvocationTargetException {
```
The already have a test that covers this line according to the online demo. Might be worth leaving out.

petersear replied:
Removed.

paulkennethkent commented on src/test/java/com/github/pagehelper/util/ExecutorUtilTest.java:
```diff
@@ -0,0 +1,64 @@
+   */
+  @PrepareForTest({Configuration.class})
+  @Test
+  public void getExistedMappedStatementInputNotNullNotNullOutputNotNull()
```
The already have a test that covers this line according to the online demo. Even though there is a single line missing, I think this is down to bluecov.

petersear replied:
Agreed. Removed.

paulkennethkent reviewed (changes requested):
Remove tests that are more than likely covered by existing tests and change pom.xml

peterschrammel reviewed (commented):
None of these tests significantly above borderline, unfortunately.

peterschrammel commented:
What's the existing coverage?

petersear commented:
These tests mostly come from the invalid bucket from test-validator which identified 9 "valid" tests out of 102.
One of those failed to compile and 3 others appear to cover lines which are already covered (as pointed out by @paulkennethkent). That was the reason I rejected some of the other "valid" tests.
Existing coverage was calculated by platform to be around 65%.

Commit:
Add Diffblue Deeptest tests

The classes covered by these tests are:

        OrderByParser
        QueryInterceptor
        SimpleCache

In order to ensure these tests run successfully, a selection of dependencies have been added to the POM file (Powermock, Mockito & Diffblue's own Deeptest Utils)

paulkennethkent commented:
Close PR, no good tests this time around 👎

paulkennethkent reviewed (commented):
👎

petersear commented:
No suitable tests found. Closing this PR.
